src-tauri/src/output_loop.rs

```diff
@@ -0,0 +1,329 @@
+                        }
+                    }
+                }
+                OutputType::Wled { ip_address } => {
+                    // Render WLED
+                    match scene::render_scene_wled(output_id, system_t, frame) {
+                        Ok(wled_data) => {
+                            // Output via WLED
+                            let wled = wled_state.lock().await;
+                            if let Err(e) = wled.output_wled_internal(ip_address, &wled_data) {
+                                log::error!("Failed to output WLED: {}", e);
+                            }
+                        }
+                        Err(e) => {
+                            log::error!("Failed to render WLED for output {}: {}", output_id, e);
+                        }
+                    }
+                }
+            }
+
+            frame = frame.wrapping_add(1);
+
+            // Sleep to maintain target FPS
+            let elapsed = loop_start.elapsed();
+            if elapsed < frame_duration {
+                tokio::time::sleep(frame_duration - elapsed).await;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[tauri::command]
+pub async fn start_output_loop(
+    manager: State<'_, Arc<TokioMutex<OutputLoopManager>>>,
+    serial_state: State<'_, Arc<TokioMutex<SerialState>>>,
+    sacn_state: State<'_, Arc<TokioMutex<SacnState>>>,
+    wled_state: State<'_, Arc<TokioMutex<WledState>>>,
+    output_id: String,
+    output_type: String,
+    universe: Option<u16>,
+    ip_address: Option<String>,
+    target_fps: u32,
+) -> Result<(), String> {
+    let output_id_u64 = output_id
+        .parse::<u64>()
+        .map_err(|e| format!("Invalid output_id: {}", e))?;
+
+    let output_type = match output_type.as_str() {
+        "serial" => OutputType::Serial,
+        "sacn" => OutputType::Sacn {
+            universe: universe.ok_or("universe required for sACN")?,
+            ip_address: ip_address.ok_or("ip_address required for sACN")?,
+        },
+        "wled" => OutputType::Wled {
+            ip_address: ip_address.ok_or("ip_address required for WLED")?,
+        },
+        _ => return Err(format!("Unknown output type: {}", output_type)),
+    };
+
+    let manager = manager.lock().await;
+    manager
+        .start_loop(
+            output_id_u64,
+            output_type,
+            target_fps,
+            serial_state.inner().clone(),
+            sacn_state.inner().clone(),
+            wled_state.inner().clone(),
+        )
+        .await
+}
+
+#[tauri::command]
+pub async fn stop_output_loop(
+    manager: State<'_, Arc<TokioMutex<OutputLoopManager>>>,
+    output_id: String,
+) -> Result<(), String> {
+    let output_id_u64 = output_id
+        .parse::<u64>()
+        .map_err(|e| format!("Invalid output_id: {}", e))?;
+
+    let manager = manager.lock().await;
+    manager.stop_loop(output_id_u64).await
+}
+
+#[tauri::command]
+pub async fn rebuild_output_loops(
```

Comment:
Instead of rebuilding all loops, can we just check if the state has changed and only restart the loop if a change is detected?